packages/vanilla/src/complex/array/ArrayControl.tsx

```diff
@@ -44,38 +44,41 @@ export const ArrayControl = ({
     () => findUISchema(uischemas, schema, uischema.scope, path, undefined, uischema, rootSchema),
     [uischemas, schema, uischema.scope, path, uischema, rootSchema]
   );
+  const isValid = errors.length === 0;
+  const divClassNames = [validationClass]
+    .concat(isValid ? '' : getStyleAsClassName('array.table.validation.error'))
+    .join(' ');  
   return (
     <div className={classNames.wrapper}>
-      <fieldset className={classNames.fieldSet}>
-        <legend>
-          <button
+      <header>
+        <label className={'array.label'}>{label}</label>
```

Comment:
We should probably scope this to `array.control.label` so that all classes are unique and don't interfere, e.g. a customization for the table renderer should not also modify the control variant.

Reply:
I have now implemented your suggestions